### PR TITLE
Async mode set to threading in debug mode (#416)

### PR DESCRIPTION
**Pull request by dinhlongviolin1**
Undirectly solve issue #416. Userman will need to be updated to document the change.

### CODE REVIEW

File: src/taipy/gui/gui.py [hunk 1]
```diff
@@ -1493,6 +1493,9 @@ def run(
         # Load application config from multiple sources (env files, kwargs, command line)
         self._config._build_config(run_root_dir, self.__env_filename, kwargs)
 
+        if app_config["debug"]:
+            async_mode = "threading"
```
Comment (FredLL-Avaiga) [inline src/taipy/gui/gui.py:1497]: Add a warning if async_mode is not threading?

**Review by FredLL-Avaiga — APPROVED** [review]
Simple enough

File: src/taipy/gui/gui.py (hunk 1 above)
Comment (FabienLelaquais) [reply src/taipy/gui/gui.py:1497]: You mean if it is forced to something else?
That would make sense...

**Review by FabienLelaquais — APPROVED** [review]
Good to go!

**Pull request merged**